src/connection/sqlJob.ts

```diff
@@ -274,10 +274,9 @@ export class SQLJob {
     return rpy;
   }
 
-  async setSelfCodes(codes: string[]) {
-    const signedCodes: String[] = codes.map(code => [code, `-${code}`]).flat();
+  async setSelfCodes(codes: string) {
     try {
-      const query: string = `SET SYSIBMADM.SELFCODES = SYSIBMADM.VALIDATE_SELF('${signedCodes.join(', ')}')`
+      const query: string = `SET SYSIBMADM.SELFCODES = '${codes}'`
       await this.query<any>(query).run();
```

Comment:
Consider using the second parameter to `query`, which accepts binding parameters in place of `?` in the statement here.

Reply:
We spoke with Jesse. There is a bug in the SC that stops this from working.